sdk/include/aditof/depth_sensor_interface.h

```diff
@@ -104,6 +104,14 @@ class DepthSensorInterface {
      */
     virtual aditof::Status getFrame(uint16_t *buffer) = 0;
 
+    /**
+     * @brief Request timestamp of the requested frame
+     * @param timestamp - a valid location where the new timestamp should be stored.
+     * @return Status
+     */
+
+    virtual aditof::Status getFrameTimestamp(long long &timestamp) = 0;
```

Comment:
I would do things differently. I would modify the getFrame() so that it retrieves the timestamp as well (besides the buffer). This way we get all we need (buffer, timestamp) in a single transaction. I would even make the second parameter a structure (let's say bufferInfo declared maybe in sensor_definitions.h) which would hold the timestamp and could hold other info in the future.
It's not wrong having getFrameTimestamp() method but it adds confusion. It raises questions like: for which frame does it return the timestamp? is it the last one? what if I call this before getFrame()? what if I close the sensor, can I still get the timestamp? what if the first call to getFrame() fails (maybe a timeout) and the second call works fine, does it get the right timestamp(for the second frame)? What if I call getFrame() and in a separate thread I call getFrameTimestamp()?
Any thoughts? @SeptimiuVana @acozma

Reply:
Well we will get the timestamp of the last frame ( if it was successfully requested) because it is set when we request a frame. The problem might appear (as you mentioned above) when we try to get a timestamp if we do not request a frame.(It will throw an error with no timestamp). 
It should be better if we get both the frame and the timestamp in getFrame() but we can add the timestamp as an default parameter.